zerver/lib/narrow_predicate.py

```diff
@@ -66,6 +66,8 @@ def satisfies_operator(*, operator: str, operand: str) -> bool:
                 topic_name = get_topic_from_message_info(message)
                 if not topic_name.startswith(RESOLVED_TOPIC_PREFIX):
                     return False
+            elif operator == "is" and operand == "followed" and message["type"] != "stream":
+                return False
```

Comment:
This seems to not actually check if the message is in a topic that the current user follows.

Reply:
This logic was written by @nimishmedatwal. 

From what I understand, `build_narrow_predicate` is used in `tornado` to decide which event updates to send to the client. We're not passing the user profile in this function so it'll require a lot of changes to actually check if the user is following a topic.

By default, all events for new messages will be sent. This check filters out only the events with `type == stream` and only sends those events in the case of `is followed`.

So, this means that even though this check is incorrect, it will not produce undesired results.

Reply:
On further testing, I've noticed that no matter what narrow the user is on, the corresponding `narrow` in `build_narrow_predicate` and `ClientDescriptor` in `event_queue.py` which calls `build_narrow_predicate` is always `[ ]`.

This seems like a bug? Or is there something I'm missing?

Reply:
This refers to a part of the API for a single-narrow Zulip feed. 

https://zulip.com/api/register-queue#parameter-narrow

You can use the example API client [here](https://github.com/zulip/python-zulip-api/blob/main/zulip/zulip/examples/get-messages) to mess with it. I expect it to be somewhat rare for someone to use this API with anything but stream/channel narrows. But I think the correct answer is that this function should be rejecting this particular operator, because the `build_narrow_predicate` function cannot be correctly implemented without access to the database/user, like the other unsupported operators.

```
def check_narrow_for_events(narrow: Collection[NarrowTerm]) -> None:                                    
    supported_operators = [*channel_operators, "topic", "sender", "is"]                                 
    for narrow_term in narrow:                                                                          
        operator = narrow_term.operator                                                                 
        if operator not in supported_operators:                                                         
            raise JsonableError(_("Operator {operator} not supported.").format(operator=operator))      
```

Reply:
Ok, I've modified `check_narrow_for_events` to reject `is:followed`.